js/install.js

```diff
@@ -0,0 +1,305 @@
+
+  if (incorrectTotal || !shouldRenderProgressBar()) {
+    return {
+      tick: () => {},
+    };
+  }
+
+  const logStream = getLogStream('stdout');
+
+  if (logStream.isTTY) {
+    return new ProgressBar(`fetching ${name} :bar :percent :etas`, {
+      complete: '█',
+      incomplete: '░',
+      width: 20,
+      total,
+    });
+  }
+
+  let pct = null;
+  let current = 0;
+  return {
+    tick: (length) => {
+      current += length;
+      const next = Math.round((current / total) * 100);
+      if (next > pct) {
+        pct = next;
+        logStream.write(`fetching ${name} ${pct}%\n`);
+      }
+    },
+  };
+}
+
+function npmCache() {
+  const env = process.env;
+  return (
+    env.npm_config_cache ||
+    env.npm_config_cache_folder ||
+    env.npm_config_yarn_offline_mirror ||
+    (env.APPDATA ? path.join(env.APPDATA, 'npm-cache') : path.join(os.homedir(), '.npm'))
+  );
+}
+
+function getCachedPath(url) {
+  const digest = crypto.createHash('md5').update(url).digest('hex').slice(0, 6);
+
+  return path.join(
+    npmCache(),
+    'sentry-cli',
+    `${digest}-${path.basename(url).replace(/[^a-zA-Z0-9.]+/g, '-')}`
+  );
+}
+
+function getTempFile(cached) {
+  return `${cached}.${process.pid}-${Math.random().toString(16).slice(2)}.tmp`;
+}
+
+function validateChecksum(tempPath, name, logger) {
+  let storedHash;
+  try {
+    const checksums = fs.readFileSync(path.join(__dirname, '../checksums.txt'), 'utf8');
+    const entries = checksums.split('\n');
+    for (let i = 0; i < entries.length; i++) {
+      const [key, value] = entries[i].split('=');
+      if (key === name) {
+        storedHash = value;
+        break;
+      }
+    }
+  } catch (e) {
+    logger.log(
+      'Checksums are generated when the package is published to npm. They are not available directly in the source repository. Skipping validation.'
+    );
+    return;
+  }
+
+  if (!storedHash) {
+    logger.log(`Checksum for ${name} not found, skipping validation.`);
+    return;
+  }
+
+  const currentHash = crypto.createHash('sha256').update(fs.readFileSync(tempPath)).digest('hex');
+
+  if (storedHash !== currentHash) {
+    fs.unlinkSync(tempPath);
+    throw new Error(
+      `Checksum validation for ${name} failed.\nExpected: ${storedHash}\nReceived: ${currentHash}`
+    );
+  } else {
+    logger.log('Checksum validation passed.');
+  }
+}
+
+function checkVersion() {
+  return helper.execute(['--version']).then((output) => {
+    const version = output.replace('sentry-cli ', '').trim();
+    const expected = process.env.SENTRYCLI_LOCAL_CDNURL ? 'DEV' : pkgInfo.version;
+    if (version !== expected) {
+      throw new Error(`Unexpected sentry-cli version "${version}", expected "${expected}"`);
+    }
+  });
+}
+
+function downloadBinary(logger = ttyLogger) {
+  if (process.env.SENTRYCLI_SKIP_DOWNLOAD === '1') {
+    logger.log(`Skipping download because SENTRYCLI_SKIP_DOWNLOAD=1 detected.`);
+    return;
+  }
```

Comment:
this should return Promise. it makes error on flag 1

Reply:
Ah right. Good catch!